From: Ubuntu Developers <ubuntu-devel-discuss@lists.ubuntu.com>
Date: Mon, 15 Jun 2026 13:45:00 +0100
Subject: Enable full LTO and opt-level z in the release profile

Set lto = true and opt-level = "z" in the release profile to
enable full link-time optimization and optimize aggressively for binary
size. This reduces the size of the resulting shared libraries and
executables.

Origin: vendor, Ubuntu
Forwarded: not-needed
Applied-Upstream: not-yet
Last-Update: 2026-06-15
---
 Cargo.toml | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/Cargo.toml b/Cargo.toml
index db81620..874736b 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -67,6 +67,10 @@ unused_extern_crates = "warn"
 unused_import_braces = "warn"
 unused_qualifications = "warn"
 
+[profile.release]
+lto = true
+opt-level = "z"
+
 [profile.dist]
 inherits = "release"
 lto = "thin"
